packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -30,39 +69,46 @@ const propTypes = {
    * An ISO 8601 string representation of the time value in the input.
    */
   value: PropTypes.string,
+  /**
+   * Type of time input to initialize. Must be '24' or '12'
+   */
+  variant: PropTypes.oneOf([FORMAT_12_HOUR, FORMAT_24_HOUR]),
 };
 
 const defaultProps = {
-  inputAttributes: undefined,
+  inputAttributes: {},
+  minuteAttributes: {},
+  hourAttributes: {},
   name: null,
   onBlur: null,
   onChange: null,
   value: undefined,
+  variant: FORMAT_24_HOUR,
 };
 
-const inputType = {
-  HOUR: 0,
-  MINUTE: 1,
-};
-
-const keyCodes = {
-  BACKSPACE: 8,
-  ARROWLEFT: 37,
-  ARROWUP: 38,
-  ARROWRIGHT: 39,
-  ARROWDOWN: 40,
-  DELETE: 46,
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
 };
 
 class TimeInput extends React.Component {
-  constructor(props) {
+  constructor(props, context) {
     super(props);
 
-    this.state = {
-      hour: TimeUtil.splitHour(props.value),
-      minute: TimeUtil.splitMinute(props.value),
-      isFocused: false,
-    };
+    let value = this.props.value;
+
+    if (value && !validDateRegex.test(value)) {
+      if (process.env !== 'production') {
+        // eslint-disable-next-line
+        console.warn(`Invalid time ${value} pass to terra-time-picker and has been ignored. Format of time must be hh:mm.`);
```

Comment:
nit: Might reword to `An invalid time value, ${value}, has been passed to the terra-time-picker. This value has been ignored and will not be rendered. Time values must be in hh:mm format.`

Reply:
https://github.com/cerner/terra-core/pull/1027/commits/1a87f172650535bec63b29cb9aa8ee3ce6810f0a